nestedtensor/csrc/python_functions.cpp

```diff
@@ -187,14 +187,81 @@ void add_functions(
 
   m.def("interpolate", 
         [](THPNestedTensor input,
-           c10::optional<THPArrayRef<int64_t>> size,
+           c10::optional<std::vector<std::vector<int64_t>>> size,
            c10::optional<THPArrayRef<double>> scale_factor,
            c10::optional<std::string> mode,
            c10::optional<bool> align_corners,
            c10::optional<bool> recompute_scale_factor) {
              if (size.has_value()) {
                return THPNestedTensor(interpolate(input.data().contiguous(), 
-                                                  size.value().extract<2>(),
+                                                  size.value(),
+                                                  c10::nullopt, 
+                                                  mode,
+                                                  align_corners));
+             }
+
+             if (scale_factor.has_value()) {
+               return THPNestedTensor(interpolate(input.data().contiguous(), 
+                                                  c10::nullopt,
+                                                  scale_factor.value().extract<2>(), 
+                                                  mode,
+                                                  align_corners));
+             }
+
+             throw "Either size or scale factor have to be passed.";
+        },
+        py::arg("input"),
+        py::arg("size") = nullptr,
+        py::arg("scale_factor") = nullptr,
+        py::arg("mode") = "nearest",
+        py::arg("align_corners") = false,
+        py::arg("recompute_scale_factor") = false);
+
+  m.def("interpolate", 
+        [](THPNestedTensor input,
+           c10::optional<std::vector<int64_t>> size,
+           c10::optional<THPArrayRef<double>> scale_factor,
+           c10::optional<std::string> mode,
+           c10::optional<bool> align_corners,
+           c10::optional<bool> recompute_scale_factor) {
+             if (size.has_value()) {
```

Comment:
Are these if/else branches still necessary even with your new signature?

Or can you create an optional ```std::vector<std::vector<int64_t>> sizes``` which you set to a value if ```size``` is defined and then otherwise always pass ```scale_factor.value().extract<2>()``` as a second argument?

Maybe we'll need to make extract<2> return an optional std vector or create a extract_opt<2>

Reply:
only size OR scale factor can be passed, so you cant pass both.

Reply:
so you cant pass both cause you dont know what will be checked first, size.has_value() or scalar_factor.has_value()